### PR TITLE
Enables multi-GPU Batch Norm, connects training of Batch Norm parameters to current optimizer, and APIifies incoming (pre-BN) versus outgoing (post-BN) units and deltas.

### PR DESCRIPTION
**Pull request by scottlegrand**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by slegrandA9** [comment]
Do not approve this pull request yet.  Edward and I need to make sure it's correct.  It may indeed already be so, but we need to make sure...

**Comment by ekandrotA9** [comment]
Reviewed the code and reran cifar-10 with Batch Norm - all looked good.
